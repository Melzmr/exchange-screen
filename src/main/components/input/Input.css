.input_base {
    font-size: 36px;
    width: 100%;
    padding: 0;
    text-align: right;
}
