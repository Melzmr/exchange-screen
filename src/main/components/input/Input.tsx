import React, {CSSProperties} from 'react';
import './Input.css';

export interface IInputProps {
  value: string | '';
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  style: CSSProperties;
}

export const Input = ({value, onChange, style}: IInputProps) => (
    <input
        className="input_base"
        type="text"
        style={style}
        value={value}
        onChange={onChange}
        inputMode="numeric"
        pattern="[0-9]*"
    />
)

Input.displayName = 'Input';
