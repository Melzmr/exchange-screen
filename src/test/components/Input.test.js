import React from 'react';
import {shallow} from 'enzyme';
import {Input} from '../../main/components/input/Input';

describe('Input', () => {
  let comp;
  let fn;

  beforeEach(() => {
    fn = jest.fn();
    comp = (
        <Input
            onChange={fn}
            style={{}}
            value={'100.00'}
        />
    );
  });

  it('renders Input component', () => {
    const wrapper = shallow(comp);
    expect(wrapper).toMatchSnapshot();
  });

});


