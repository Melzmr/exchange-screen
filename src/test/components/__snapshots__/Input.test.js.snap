// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders Input component 1`] = `
<input
  className="input_base"
  inputMode="numeric"
  onChange={[MockFunction]}
  pattern="[0-9]*"
  style={Object {}}
  type="text"
  value="100.00"
/>
`;
